pkg/controller/kieapp/shared/utils.go

```diff
@@ -174,3 +180,51 @@ func GetNamespacedName(object metav1.Object) types.NamespacedName {
 		Namespace: object.GetNamespace(),
 	}
 }
+
+func GetConsoleYAMLSample(res resource.KubernetesResource) (*consolev1.ConsoleYAMLSample, error) {
+	snippetStr := res.GetAnnotations()["consoleSnippet"]
```

Comment:
I'm not clear on what this does and what purpose it serves. So it's ok to leave it out, but if the operator passes in a yaml with consoleSnippet set to false, we return an error instead of a ConsoleYAMLSample? Not sure that's necessary, or why an operator would pass it in, if they've marked as false / not a snippet. I would just remove this whole part.

Reply:
Since now we by default try our best to create the ConsoleYAMLSample, this error return is not necessary, and we should just give a default value (false: not a snippet).   But we should still have consoleSnippet to allow user to turn the entire yaml as snippet: set to true.
I'd do like this:
	var snippet bool = false
          .....
		tmp, err := strconv.ParseBool(snippetStr)
		if err == nil {
			snippet = tmp
		}

Reply:
I don't understand!

Reply:
by default this yaml file is not a snippet.  If user wants to mark it as snippet, he gives the consoleSnippet: true. But if the string value has typo: not parsable, we keep it as false.

Reply:
oh, I misunderstood this whole thing. What is a snippet? That means the yaml is not your whole CR? Is there a difference in behavior, perhaps in that it doesn't completely replace your yaml when clicked?

Reply:
Correct, different behavior, doesn't completely replace when clicked.
another tab will be created and named "Snippet", normal Yamls have "Samples" tab.